org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/builder/MavenBuilderImpl.java

```diff
@@ -219,7 +225,11 @@ private boolean hasRelevantDelta(IMavenProjectFacade projectFacade, IResourceDel
         IPath fullPath = delta.getFullPath();
         if(buildOutputLocation.isPrefixOf(fullPath)) {
           //anything in the build output is not interesting for a change as it is produced by the build
-          //lets see if there are more interesting parts...
+          // ... unless a classpath resource that existed before has been deleted, possibly by another builder
+          if(!resource.exists() && isOutputOrTestOutput.test(fullPath)) {
```

Comment:
```suggestion
          if(isOutputOrTestOutput.test(fullPath) && !resource.exists()) {
```
checking the prefix is probably a bit cheaper than checking all resources for existence.